docs/components.md

```diff
@@ -0,0 +1,94 @@
+    - `bash ./<nodepool_name>/create_nodepool.sh` will create the nodepool as configured
+    - Argument keys are lower case, underscore separated version of the [gcloud container node-pools create](https://cloud.google.com/sdk/gcloud/reference/beta/container/node-pools/create) command
+    - If a `-f` file is passed in, data are merged with `-D` values ovveriding the file values
+    - Example: 
+      ```
+      pentagon add gcp.nodepool -D name="pentagon-1-nodepool" -D project="pentagon" -D zone="us-central1-a" -D additional_zones="us-central1-b,us-central1-b" -D machine_type="n1-standard-64" --enable-autoscaling
+      ```
+
+- get: 
+    - Creates `./<nodepool_name>/create_nodepool.sh` by querying the state of an existing cluster nodepool and parsing values. For when you have an existing cluster that you want to capture its configuration.
+    - Creates `./<nodepool_name>/create_nodepool.sh` 
+    - `bash ./<nodepool_name>/create_nodepool.sh` will result in an error indicating the cluster is already present.
+    - Argument keys are lower case, underscore separated version of the [gcloud container node-pools describe](https://cloud.google.com/sdk/gcloud/reference/beta/container/node-pools/describe) command
+    - If a `-f` file is passed in, data are merged with `-D` values ovveriding the file values
+    - If `name` is omitted it will act on all nodepool in the cluster
+    - Example:
+        ```
+        pentagon get gcp.nodepool -D project="pentagon" -D zone="us-central1-a -D cluster="pentagon-1" -D name="pentagon-1-nodepool"
+        ```
+
+
+### vpc
+
+- add: 
+    - Creates `./vpc/` directory with Terraform code for the Pentagon default AWS VPC described [here](#network).
+    - `cd ./vpc; make all` will create the vpc as describe by the arguments passed in
+    - In the normal course of using Pentagon and the infrastructure repository, it is unlikely you'll use this component.
+    - Required Arguments:
+        - vpc_name
+        - vpc_cidr_base
+        - aws_availabilty_zones
+        - aws_availability_zone_count
+        - aws_region
+        - infrastructure_bucket
+    - Example:
+        ```
+        pentagon add vpc -D vpc_name="pentagon-vpc" -D vpc_cidr_base="172.20" -D aws_availability_zones="ap-northeast-1a, ap-northeast-1c" -D aws_availability_zone_count = "2" -D aws_region = "ap-northeast-1"
+        ```
+        
+## Writing your own components
+
+Component modules must be named `pentagon<component_name>`. Classes are subclasses of the `pentagon.component.ComponentBase` class and they must be named <Component> (note the capital first letter).  The `pentagon add <component_name>` command will prefer built in components to extrenal components so ensure your component name is not already in use. The <component_name> argument can be a dot separated module path ie `gcp.cluster` where the last parameter is the lowercase class name. For example. `gcp.cluster` finds the Cluster class in the cluster module in the gcp module.
```

Comment:
s/extrenal/external/

Reply:
Can you add examples for the the naming of components to make this a bit more clear?